src/main/java/it/gov/pagopa/admissibility/service/onboarding/AuthoritiesDataRetrieverServiceImpl.java

```diff
@@ -29,101 +36,133 @@ public class AuthoritiesDataRetrieverServiceImpl implements AuthoritiesDataRetri
+                                        .build()
+                                        : Residence.builder()
+                                        .city("Roma")
+                                        .cityCouncil("Roma")
+                                        .province("Roma")
+                                        .region("Lazio")
+                                        .postalCode("00187")
+                                        .nation("Italia")
+                                        .build()
+
+                        );
+                    }
+                })
+                // BIRTHDATE
+                .doOnNext(o -> {
+                    if (onboardingRequest.getBirthDate() == null && is2retrieve(initiativeConfig, OnboardingConstants.CRITERIA_CODE_BIRTHDATE)) {
+                        int age = userIdBasedIntegerGenerator(onboardingRequest).nextInt(18, 99);
+                        onboardingRequest.setBirthDate(BirthDate.builder()
+                                .age(age)
+                                .year((LocalDate.now().getYear() - age) + "")
+                                .build());
+                    }
+                });
+    }
+
+    private Mono<OnboardingDTO> retrieveIsee(OnboardingDTO onboardingRequest, InitiativeConfig initiativeConfig) {
+        return this.retrieveUserIsee(onboardingRequest.getUserId())
+                .switchIfEmpty(mockIsee(onboardingRequest))
+                .doOnNext(m -> setIseeIfCorrespondingType(onboardingRequest, initiativeConfig, m))
+                .flatMap(m -> {
+                    CriteriaCodeConfig criteriaCodeConfig = criteriaCodeService.getCriteriaCodeConfig(OnboardingConstants.CRITERIA_CODE_ISEE);
+
+                    if (onboardingRequest.getIsee() == null) {
+                        return Mono.error(new OnboardingException(
+                                List.of(new OnboardingRejectionReason(
+                                        OnboardingRejectionReason.OnboardingRejectionReasonType.ISEE_TYPE_KO,
+                                        OnboardingConstants.REJECTION_REASON_ISEE_TYPE_KO,
+                                        criteriaCodeConfig.getAuthority(),
+                                        criteriaCodeConfig.getAuthorityLabel(),
+                                        "ISEE non disponibile"
+                                )),
+                                "User having id %s has not compatible type for initiative %s".formatted(onboardingRequest.getUserId(), initiativeConfig.getInitiativeId()))
+                        );
+                    } else {
+                        return Mono.just(onboardingRequest);
+                    }
+                });
+    }
+
+    private Mono<Map<String, BigDecimal>> mockIsee(OnboardingDTO onboardingRequest) {
+        Map<String, BigDecimal> iseeMockMap = new HashMap<>();
+        List<IseeTypologyEnum> iseeList = new ArrayList<>(Arrays.asList(IseeTypologyEnum.values()));
```

Comment:
if not modified, you could work with the immutable list obtained
```suggestion
        List<IseeTypologyEnum> iseeList = Arrays.asList(IseeTypologyEnum.values());
```